plugins/woocommerce/includes/wc-update-functions.php

```diff
@@ -2348,7 +2348,11 @@ function wc_update_630_create_product_attributes_lookup_table() {
 	$data_store       = wc_get_container()->get( LookupDataStore::class );
 	$data_regenerator = wc_get_container()->get( DataRegenerator::class );
 
-	if ( $data_store->check_lookup_table_exists() ) {
+	/**
+	 * If the table exists and contains data, it was manually created by user before the migration ran.
+	 * If the table exists but is empty, it was likely created right now via dbDelta, so a table regenerations is needed.
+	 */
+	if ( $data_store->check_lookup_table_exists() && $data_store->lookup_table_has_data() ) {
```

Comment:
If we're not checking other edge cases like user deleting the table, then it seems unnecessary to check whether the table was added manually as well. Or should this handle stores that activated this feature in its previous version when it didn't include indices yet?

Reply:
Yes, exactly. This covers all the cases: table not existing, table existing with or without data, table existing with or without indices.

Reply:
Just for completeness, pasting it here as well:
it seems to me that dbDelta takes care of the indices

This is what I did:
1. Installed WC 6.2
2. Added products
3. Enabled the lookup table
4. Created a zip pkg from your branch but commented out $data_regenerator->maybe_create_table_indices();
5. Ran the update
6. Checked the indices were in place after the update

So unless I'm missing a scenario, `maybe_create_table_indices` method might be unnecessary?